src/utils/hooks/useMultiClusterNamespaces.ts

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { useMemo } from 'react';
+
+import { modelToGroupVersionKind, NamespaceModel } from '@kubevirt-ui/kubevirt-api/console';
+import { getName } from '@kubevirt-utils/resources/shared';
+import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+import { OBJECTS_FETCHING_LIMIT } from '@virtualmachines/utils';
+
+import useKubevirtWatchResource from './useKubevirtWatchResource';
+
+type MultiClusterNamespaces = () => [Record<string, string[]>, boolean, any];
+
+const useMultiClusterNamespaces: MultiClusterNamespaces = () => {
+  const [namespaceData, loaded, error] = useKubevirtWatchResource<K8sResourceCommon[]>({
+    groupVersionKind: modelToGroupVersionKind(NamespaceModel),
+    isList: true,
+    limit: OBJECTS_FETCHING_LIMIT,
+  });
+
+  const namespaceNameByClusterRecord = useMemo(() => {
+    const namespaceClusterMap: Record<string, string[]> = {};
+    namespaceData?.forEach((resource) => {
+      namespaceClusterMap[resource.cluster]
```

Comment:
`resource.cluster` will be undefined in the single-cluster case. Need to handle it.